addons/l10n_ar_base/models/res_partner.py

```diff
@@ -0,0 +1,126 @@
+                cuit[0:2], cuit[2:10], cuit[10:])
+
+    @api.depends('l10n_ar_id_number', 'l10n_ar_id_category_id')
+    def _compute_l10n_ar_cuit(self):
+        """ We add this computed field that returns cuit or nothing ig this one
+        is not set for the partner. This validation can be also dony by calling
+        cuit_required() method that returns the cuit nombre of error if this
+        one is not found.
+        """
+        for rec in self:
+            # If the partner is outside Argentina then we return the defined
+            # country cuit defined by AFIP for that specific partner
+            if rec.l10n_ar_id_category_id.afip_code != 80:
+                country = rec.country_id
+                if country and country.code != 'AR':
+                    if rec.is_company:
+                        rec.l10n_ar_cuit = country.l10n_ar_cuit_juridica
+                    else:
+                        rec.l10n_ar_cuit = country.l10n_ar_cuit_fisica
+                continue
+            if rec.l10n_ar_id_category_id.afip_code == 80:
+                rec.l10n_ar_cuit = rec.l10n_ar_id_number
+
+    @api.constrains('l10n_ar_id_number', 'l10n_ar_id_category_id')
+    def check_vat(self):
+        """ Update the the vat field using the information we have from
+        l10n_ar_id_number and l10n_ar_id_category_id fields
+        """
+        for rec in self:
+            if rec.l10n_ar_id_number and rec.l10n_ar_id_category_id and \
+               rec.l10n_ar_id_category_id.afip_code == 80:
+                rec.vat = 'AR' + rec.l10n_ar_id_number
+
+    @api.constrains('l10n_ar_id_number', 'l10n_ar_id_category_id')
+    def check_id_number_unique(self):
+        """ Taking into account the company's general settings it will check
+        that if the identification number we are trying to use is already set
+        in another partner.
+        """
+        if not safe_eval(self.env['ir.config_parameter'].sudo().get_param(
+                "l10n_ar_base.unique_id_numbers", 'False')):
+            return True
+        for rec in self:
+            # We allow same number in related partners
+            related_partners = rec.search([
+                '|', ('id', 'parent_of', rec.id),
+                ('id', 'child_of', rec.id)])
+            same_id_numbers = rec.search([
+                ('l10n_ar_id_number', '=', rec.l10n_ar_id_number),
+                ('l10n_ar_id_category_id', '=', rec.l10n_ar_id_category_id.id),
+                ('id', 'not in', related_partners.ids),
+            ])
+            if same_id_numbers:
+                raise ValidationError(_(
+                    'Identification number must be unique per Identification'
+                    ' category!\nSame number is only allowed for partners with'
+                    ' parent/child relation\n\n Already using this'
+                    ' number: ') + ', '.join(same_id_numbers.mapped('name'))
+                )
```

Comment:
In saas 12.3, we added a warning on the partner when it has the same VAT as another one.  If this sets the vat, the check could be the same.

Reply:
We do this not only for vat (CUIT) but also for all the other types of documents. Also, some customers wants an strong constraint and not just a warning. But if you prefer we can remove this as is not mandatory and keep this on our modules.

Reply:
remove this and check if odoo vat warning of odoo 12.3 works with our identification number (perhaps we can make something similar to odoo)